File: CSharpFundamentals/CSharpAdvanced/StacksAndQueues_Lab/HotPotato.cs
using System;
using System.Collections.Generic;

namespace HotPotato
{
    class HotPotato
    {
        static void Main(string[] args)
        {
            string[] splitted = Console.ReadLine().Split();
            int index = int.Parse(Console.ReadLine());
            Queue<string> queue = new Queue<string>(splitted);
            while (queue.Count > 1)
            {
                for (int i = 1; i < index; i++)
                {
                    queue.Enqueue(queue.Dequeue());
                }
                Console.WriteLine($"Removed {queue.Dequeue()}");
            }
            Console.WriteLine($"Last is {queue.Dequeue()}");
        }
    }
}

File: JavaFundamentals/JavaAdvanced/Stacks-and-queues-lab/src/HotPotato.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.ArrayDeque;
import java.util.Collections;

public class HotPotato {
    public static void main(String[] args) throws IOException {
        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));

        String[] names = reader.readLine().split("\\s+");
        int count = Integer.parseInt(reader.readLine());

        ArrayDeque<String> childrenQueue = new ArrayDeque<>();
        Collections.addAll(childrenQueue, names);

        while (childrenQueue.size() > 1){
            for (int i = 0; i < count - 1; i++) {
                String leftChildren = childrenQueue.poll();
                childrenQueue.offer(leftChildren);
            }
            String removed = childrenQueue.poll();
            System.out.printf("Removed %s%n", removed);
        }

        String winner = childrenQueue.poll();

        System.out.printf("Last is %s", winner);
    }
}
